### PR TITLE
feat: upgrade pyjwt to version 2.1.0

### PR DESCRIPTION
**Pull request by iamsobanjaved**
- Resolved all `pyjwt` version 2.0 breaking changes
- Overrriden a few global constraints by downloading the common constraint file
- Upgraded `edx-drf-extensions` to the latest version
- Upgraded `edx-auth-backends` to the latest version

For more info about pyjwt changes: https://pyjwt.readthedocs.io/en/stable/changelog.html

NOTE: This PR need proper testing on stage. Please make sure deployment contains only these changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,7 +45,21 @@ production-requirements: ## Install Python and JS requirements for production
...
 
-upgrade:
+COMMON_CONSTRAINTS_TXT=requirements/common_constraints.txt
+.PHONY: $(COMMON_CONSTRAINTS_TXT)
+$(COMMON_CONSTRAINTS_TXT):
+	wget -O "$(@)" https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt || touch "$(@)"
+
+
+upgrade: $(COMMON_CONSTRAINTS_TXT)
+	sed 's/pyjwt\[crypto\]<2.0.0//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (iamsobanjaved) [inline Makefile:55]: We are creating a local copy of the common constraints file and override/remove the global constraint which we don't need. This approach is used temporarily and will be improved in https://openedx.atlassian.net/browse/BOM-2721

File: requirements/local.txt [hunk 2]
```diff
@@ -282,7 +277,7 @@ edx-rest-api-client==5.4.0
     #   taxonomy-connector
 edx-sphinx-theme==3.0.0
     # via -r requirements/docs.in
-elasticsearch==7.13.4
+elasticsearch==7.14.0
```
Comment (waheedahmed) [inline requirements/local.txt]: `7.14.0` has some breaking changes, it broke the production when we deployed last time. I'm not sure why it's upgraded to `7.14.0` even we have a constraint for it https://github.com/edx/course-discovery/pull/3086/files#diff-86d5fe588ff2fc7dccb1f4cdd8019d4473146536e88d7a9ede946ea962a91acbR10?

**Review by waheedahmed — CHANGES_REQUESTED** [review]
One concern otherwise looks good.

File: requirements/local.txt (hunk 2 above)
Comment (awais786) [reply requirements/local.txt]: I think PR created before the constraint.

**Review by awais786 — APPROVED** [review]

File: requirements/common_constraints.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# 5.0.0+ of social-auth-app-django requires social-auth-core>=4.1.0
+social-auth-app-django<5.0.0
+
+# latest version requires PyJWT>=2.0.0 but drf-jwt requires PyJWT[crypto]<2.0.0,>=1.5.2.
+# See  comment.
+
+
+# elasticsearch>=7.14.0 includes breaking changes in it which caused issues in discovery upgrade process.
+# elastic search changelog: https://www.elastic.co/guide/en/enterprise-search/master/release-notes-7.14.0.html
+elasticsearch<7.14.0
```
Comment (waheedahmed) [inline requirements/common_constraints.txt:43]: This is already in place inside `requirements/constraints.txt`.

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**